tests/test_action_init.py

```diff
@@ -20,6 +21,7 @@ def test_run(self, mock_init, mock_check_result, mock_chdir):
         test_plan_path = "/terraform"
         test_terraform_exec = "/usr/bin/terraform"
         test_backend = {'path': '/terraform/terraform.tfstate'}
+        test_upgrade = True
 
```

Comment:
you should probably have some extra tests in here that passes in `None` and `False` for the `upgraded` to ensure proper behavior.

The idea here is the we want to verify the code works under all circumstances